src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -539,6 +511,50 @@ private static void ExecuteAndCompleteQuery(QueryExecuteParams executeParams, Re
             }
         }
 
+        // Internal for testing purposes
+        internal string GetSqlText(ExecuteRequestParamsBase request)
+        {
+            Type requestType = request.GetType();
+
+            // If it is a document selection, we'll retrieve the text from the document
+            if (requestType == typeof(ExecuteDocumentSelectionParams))
```

Comment:
Agreed, I think I wrote it like this assuming there was the ability to do a switch on types. I'll modify this as per your suggestion.